test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -21,222 +21,216 @@
-                                service
-                                "production"))))
-           (let [production-cache-id-before-marked-expired
-                 (jruby-protocol/get-environment-class-info-cache-generation-id!
-                  service
-                  "production")]
-             (jruby-protocol/mark-all-environments-expired! service)
-             (is (nil? (jruby-protocol/get-environment-class-info-tag
-                        service
-                        "production")))
-             (is (not= production-cache-id-before-marked-expired
-                       (jruby-protocol/get-environment-class-info-cache-generation-id!
-                        service
-                        "production")))
-             (is (nil? (jruby-protocol/get-environment-class-info-tag service
-                                                                      "test")))
-             (is (not= test-cache-id-after-first-update
-                       (jruby-protocol/get-environment-class-info-cache-generation-id!
-                        service
-                        "test")))))
-         (testing (str "when all environments expired between get and set "
-                       "for environment that did not previously exist")
-           (let [staging-cache-id-before-marked-expired
-                 (jruby-protocol/get-environment-class-info-cache-generation-id!
-                  service
-                  "staging")
-                 _ (jruby-protocol/mark-all-environments-expired! service)
-                 staging-cache-id-after-marked-expired
-                 (jruby-protocol/get-environment-class-info-cache-generation-id!
-                  service
-                  "staging")]
-             (jruby-protocol/set-environment-class-info-tag!
-              service
-              "staging"
-              "1234staging"
-              staging-cache-id-before-marked-expired)
-             (is (= nil (jruby-protocol/get-environment-class-info-tag
-                         service
-                         "staging"))
-                 (str "Tag should not have been changed by the prior set "
-                      "since the environment was marked expired after the "
-                      "`staging-cache-id-before-marked-expired` was read"))
-             (is (= staging-cache-id-after-marked-expired
-                    (jruby-protocol/get-environment-class-info-cache-generation-id!
-                     service
-                     "staging"))
-                 (str "Cache id should not have been changed by the prior "
-                      "set since the environment was marked expired after the "
-                      "`staging-cache-id-before-marked-expired` was "
-                      "read")))))))))
+                        "set since the environment was marked expired after the "
+                        "`staging-cache-id-before-marked-expired` was "
+                        "read"))))))))))
```

Comment:
I don't really have my head wrapped around the implications of testing the etag stuff w/o real jrubies but I will trust your judgment on this commit.